liam2/alignment.py

```diff
@@ -295,11 +317,26 @@ def _display_unaligned(self, expressions, ids, columns, unaligned):
 
     def compute(self, context, score, need, filter=None, take=None, leave=None,
                 expressions=None, possible_values=None, errors='default',
-                frac_need='uniform', link=None, secondary_axis=None):
+                frac_need='uniform', link=None, secondary_axis=None,
+                method='default'):
+
+        if method not in ("default", "sidewalk"):
+            raise Exception("Method for alignment should be either 'default' "
+                            "either 'sidewalk'")
+        if need is None and method != 'sidewalk':
+            raise Exception("No default value for need if method is not sidewalk")

```

Comment:
if need can be None for sidewalk, maybe need should be made optional (None by default)